dashboard/app/models/unit_group.rb

```diff
@@ -522,6 +522,33 @@ def redirect_to_course_url(user)
     latest_assigned_version.link
   end
 
+  def can_be_instructor?(user)
+    return false if user.student?
+    return true if user.permission?(UserPermission::CODE_INSTRUCTOR) || user.permission?(UserPermission::LEVELBUILDER)
+
+    if instructor_audience == 'plc_reviewer'
+      return user.permission?(UserPermission::PLC_REVIEWER)
+    elsif instructor_audience == 'facilitator'
+      return user.permission?(UserPermission::FACILITATOR)
+    elsif instructor_audience == 'teacher'
+      return user.teacher?
+    end
+
+    false
+  end
+
+  def can_be_participant?(user)
+    if participant_audience == 'facilitator'
+      return user.permission?(UserPermission::FACILITATOR)
+    elsif participant_audience == 'teacher'
+      return user.teacher?
+    elsif participant_audience == 'student'
+      return true #if participant audience is student let anyone join
+    end
+
+    false
+  end
```

Comment:
Since these are nearly identical to above is there a way to extract the logic and re-use?

Reply:
One potential way to reduce duplication would be to have `in_participant_audience?(participant_audience)` as a method in `user.rb`. Then you could call it with either the script or unit_group participant audience like, `user.in_participant_audience?(script.participant_audience)`.

Reply:
Good idea Maureen! I'll do that

Reply:
great ideas Erin and Maureen! since the work to consolidate these methods into UnitGroup is not yet scheduled, it seems worth finding a way to deduplicate this code. I would suggest extracting the new methods into a [Concern](https://api.rubyonrails.org/classes/ActiveSupport/Concern.html), which is included by both Script and UnitGroup. The idea of using the User model would solve it, which could make sense if we want to use these concepts outside of curriculum. but until we discover that it seems like the logic would better belong closer to curriculum code.

Reply:
Alright I think I have it updated to use a Concern and working!

Reply:
I'd be interested in having a side discussion about Concerns sometime.  My experience with them is that they're brittle and difficult for someone unfamiliar with the codebase to understand but maybe I've just been using them incorrectly!

Reply:
I did read some stuff saying to never use them when I was learning about them. There were others saying they are great. 

@jamescodeorg I would be interested. What would you suggest for something like this?

Reply:
Maybe just start a thread in Slack where we can share articles and experiences?  I think that would be a good way to get some ideas on the table and give everyone time to (asynchronously) think about it.  Then, if there's interest, we can maybe get together for a meeting.

Reply:
Added in dashbaord-dev